src/backend/distributed/commands/database.c

```diff
@@ -226,6 +243,30 @@ PreprocessAlterDatabaseRefreshCollStmt(Node *node, const char *queryString,
 
 #endif
 
+
+List *
+PreprocessAlterDatabaseRenameStmt(Node *node, const char *queryString,
+								  ProcessUtilityContext processUtilityContext)
+{
+	if (!ShouldPropagate())
+	{
+		return NIL;
+	}
+
+	RenameStmt *stmt = castNode(RenameStmt, node);
+
+	EnsureCoordinator();
```

Comment:
Please move this check before the `castNode` line.

Reply:
Fixed